redisson/src/main/java/org/redisson/codec/FuryCodec.java

```diff
@@ -85,22 +84,35 @@ public FuryCodec(ClassLoader classLoader, boolean registrationRequired, Language
     private final Decoder<Object> decoder = new Decoder<Object>() {
         @Override
         public Object decode(ByteBuf buf, State state) throws IOException {
-            return fury.deserialize(FuryStreamReader.of(new ByteBufInputStream(buf)));
+            MemoryBuffer furyBuffer = MemoryUtils.wrap(buf.nioBuffer());
+            try {
+                return fury.deserialize(furyBuffer);
+            } finally {
+                buf.readerIndex(buf.readerIndex() + furyBuffer.readerIndex());
+            }
         }
     };
 
     private final Encoder encoder = new Encoder() {
         @Override
         public ByteBuf encode(Object in) throws IOException {
             ByteBuf out = ByteBufAllocator.DEFAULT.buffer();
-            try {
-                ByteBufOutputStream baos = new ByteBufOutputStream(out);
-                fury.serialize(baos, in);
-                return baos.buffer();
-            } catch (Exception e) {
-                out.release();
-                throw e;
+            MemoryBuffer furyBuffer;
+            int remainingSize = out.capacity() - out.writerIndex();
+            if (out.hasArray()) {
+                furyBuffer = MemoryUtils.wrap(out.array(), out.arrayOffset() + out.writerIndex(),
```

Comment:
in case of direct buffer, which is default mode .array() method throws UnsupportedOperationException.

ByteBufOutputStream is zero-copy

Reply:
The implementation in Fury is not zero-copy, we don't support native streaming serialization currently. Fury will serialize data into a binary, then copy it into the `OutputStream`. Fury will update written header in some cases for better space efficiency, so it can't send data chunk by chunk.

And in case of direct buffer, I added a `if (out.hasArray())` check here, `.array() method` will be invoked here only

Reply:
The deserilization support streaming natively, but serialization does not